src/OpenIDConnectClient.php

```diff
@@ -773,7 +773,6 @@ protected function requestTokens($code) {
         }
 
         if (!empty($this->getCodeChallengeMethod()) && !empty($this->getCodeVerifier())) {
-            $headers = [];
             unset($token_params['client_secret']);
```

Comment:
replace these 2 lines with the following:
```php
            if (empty($this->getClientSecret())) {
                $headers = [];
                unset($token_params['client_secret']);
            }
```

Reply:
THX - I'll test this with our keycloak setup asap ....

Reply:
Tested with keycloak :+1:

Reply:
Great to hear that it worked! :)